python/cudf/cudf/core/udf/scalar_function.py

```diff
@@ -48,22 +46,41 @@ def f(x, c, k):
     )
 
 
-def _get_scalar_kernel(sr, func, args):
-    sr_type = MaskedType(
-        string_view if sr.dtype == "O" else numpy_support.from_dtype(sr.dtype)
-    )
-    scalar_return_type = _get_udf_return_type(sr_type, func, args)
-
-    sig = _construct_signature(sr, scalar_return_type, args=args)
-    f_ = cuda.jit(device=True)(func)
-    global_exec_context = {
-        "f_": f_,
-        "cuda": cuda,
-        "Masked": Masked,
-        "_mask_get": _mask_get,
-        "pack_return": pack_return,
-    }
-    kernel_string = _scalar_kernel_string_from_template(sr, args=args)
-    kernel = _get_kernel(kernel_string, global_exec_context, sig, func)
-
-    return kernel, scalar_return_type
+class SeriesApplyKernel(ApplyKernelBase):
+    """
+    Class representing a kernel that computes the result of
+    a Series.apply operation. Expects that the user passed
+    a function that operates on an single element of the Series,
+    for example
+
+    def f(x):
+        return x + 1
+    """
+
+    @property
+    def kernel_type(self):
+        return "series_apply"
+
+    def _get_frame_type(self):
+        return MaskedType(
+            string_view
+            if self.frame.dtype == "O"
+            else numpy_support.from_dtype(self.frame.dtype)
+        )
+
+    def _get_kernel_string(self):
+        # This is the kernel string that will be compiled
+        # It is generated from the template in templates.py
+        # and is specific to the function being compiled
+        return _scalar_kernel_string_from_template(self.frame, self.args)
```

Comment:
Can we inline the function here? it's not being called anywhere else I believe, and now this function is a one-liner calling that one.

Reply:
Copy/paste error here, thanks 👍